core/src/main/java/com/microsoft/applicationinsights/internal/channel/ConfiguredTransmitterFactory.java

```diff
@@ -2,6 +2,17 @@
 
 import com.microsoft.applicationinsights.TelemetryConfiguration;
 
+import javax.annotation.Nullable;
+
 public interface ConfiguredTransmitterFactory<T> extends TransmitterFactory<T> {
-    TelemetriesTransmitter<T> create(TelemetryConfiguration configuration, String endpoint, String maxTransmissionStorageCapacity, boolean throttlingIsEnabled, int maxInstantRetries);
+    /**
+     * Either {@code configuration} or {@code endpoint} could be null, but one must be non-null.
+     * @param configuration The configuration for the current TelemetryClient
+     * @param endpoint The endpoint to use.
+     * @param maxTransmissionStorageCapacity
+     * @param throttlingIsEnabled
+     * @param maxInstantRetries
+     * @return
+     */
+    TelemetriesTransmitter<T> create(@Nullable TelemetryConfiguration configuration, @Nullable String endpoint, String maxTransmissionStorageCapacity, boolean throttlingIsEnabled, int maxInstantRetries);
```

Comment:
This is what I had at first, but it seemed easier to maintain backwards compatibility if we included it here. I'll see if I can refactor it out because I would prefer it not be there (we will have to remove it eventually anyway)